src/components/general/DatePicker.js

```diff
@@ -3,195 +3,132 @@ import PropTypes from "prop-types";
-			focused: date.length === 3
+			monthArray: Array.from({ length: 12 }, (v, k) => k + 1),
+			yearArray: Array.from({ length: 20 }, (v, k) => new Date().getFullYear() + k) // We only need about 20 years ahead... right?
 		};
 	}
 
-		static propTypes = {
-			value: PropTypes.object,
-			placeholder: PropTypes.string.isRequired,
-			onSelect: PropTypes.func.isRequired
-		}
-
-		prepend0(item) {
-			if (item < 10)
-				return "0" + item;
-
-			return item;
-		}
-
-		leapYear(year) {
-			return year % 4 == 0 && year % 100 != 0 || year % 400 == 0;
-		}
+	static propTypes = {
+		value: PropTypes.object,
+		onSelect: PropTypes.func.isRequired
+	}
 
-		update(item) {
-			const {
-				month,
-				day,
-				year
-			} = item;
-			this.props.onSelect(`${year}-${month}-${day}`);
-		}
+	/** Returns an array of days in the current state's month when called */
+	getDayArray() {
+		const { month, day, year } = this.state;
 
-		clickActionMonth(item) {
-			const {
-				day,
-				year
-			} = this.state;
-
-			item = this.prepend0(item);
-
-			this.setState({ month: item });
-			const month30 = [false, false, false, true, false, true,
-				false, false, true, false, true, false];
-
-			if (item === 2)
-				if (year != 0 && this.leapYear(year)) {
-					this.setState({ dayArr: Array.from({ length: 29 }, (v, k) => k + 1) });
-					if (day > 29) this.setState({ day: 29 });
-				}
-				else {
-					this.setState({ dayArr: Array.from({ length: 28 }, (v, k) => k + 1) });
-					if (day > 28) this.setState({ day: 28 });
-				}
-			if (month30[item]) {
-				this.setState({ dayArr: Array.from({ length: 30 }, (v, k) => k + 1) });
-				if (day > 30) this.setState({ day: 30 });
-			}
-			else if (item !== 2) {
-				this.setState({ dayArr: Array.from({ length: 31 }, (v, k) => k + 1) });
-			}
-			if (day !== "" && year !== "")
-				this.update({ day: day, month: item, year: year });
-		}
-		clickActionDay(item) {
-			const {
-				month,
-				year
-			} = this.state;
+		const daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+		const monthInt = parseInt(month);
 
-			item = this.prepend0(item);
-			this.setState({ day: item });
+		// If the state's day is more than the max days in a month, set the day to the max day
+		const setMaxDay = (maxDay) => { if (day > maxDay) this.setState({ day: maxDay }); };
 
-			if (month !== "" && year !== "")
-				this.update({ day: item, month: month, year: year });
+		if (monthInt == 2 && this.isLeapYear(year)) {
+			setMaxDay(29);
+			return Array.from({ length: 29 }, (v, k) => k + 1);
 		}
+		setMaxDay(daysInMonth[monthInt - 1]);
+		return Array.from({ length: daysInMonth[monthInt - 1] }, (v, k) => k + 1);
+	}
 
-		clickActionYear(item) {
-			const {
-				month,
-				day
-			} = this.state;
+	/**
+	 * Returns true if the input year is a leap year. Must be an int.
+	 * @param {Number} year Any year such as 2020 or 101
+	*/
+	isLeapYear(year) { return (year % 4 === 0) && (year % 100 != 0) || (year % 400 == 0) }
```

Comment:
Avoid one-line functions

Reply:
if you want it to be one line make it
```suggestion
	isLeapYear(year) => (year % 4 === 0) && (year % 100 != 0) || (year % 400 == 0)
```